udata/core/dataset/models.py

```diff
@@ -290,11 +290,19 @@ class ResourceMixin(object):
     harvest = db.EmbeddedDocumentField(HarvestResourceMetadata)
     schema = db.DictField()
 
-    created_at = db.DateTimeField(default=datetime.now, required=True)
-    modified = db.DateTimeField(default=datetime.now, required=True)
+    created_at_internal = db.DateTimeField(default=datetime.now, required=True)
+    last_modified_internal = db.DateTimeField(default=datetime.now, required=True)
     published = db.DateTimeField()  # DEPRECATED BUT LEFT FOR BACKWARDS COMPATIBILITY
     deleted = db.DateTimeField()
 
+    @property
+    def created_at(self):
+        return self.harvest.created_at if self.harvest and self.harvest.created_at else self.created_at_internal
```

Comment:
I'm scared we may have issues with naive datetimes, maybe use the helper used in last_update?

Reply:
the problem occured by comparing a naive date and an aware one. Here we are not doing any comparison.